src/write_to_file.cc

```diff
@@ -47,20 +51,22 @@ int writeValueToFile(
 }
 
 
-void writeVectorToFile(std::vector<double> v, std::string pFilename){
+void writeVectorToFile(std::vector<double> v, fs::path filename){
```

Comment:
Yes, I can comprehend your point and thought about the same thing that this var is only call-by-value. Just wanted to hear your opinion about this. Finally I am okay with the non const-ness